example/lib/edge.dart

```diff
@@ -0,0 +1,85 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_aepedge/flutter_aepedge_data.dart';
+import 'package:flutter_aepedge/flutter_aepedge.dart';
+import 'util.dart';
+
+class EdgePage extends StatefulWidget {
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<EdgePage> {
+  String _edgeVersion = 'Unknown';
+
+  List<EventHandle> _edgeEventHandleResponse = List.empty();
+  @override
+  void initState() {
+    super.initState();
+    initPlatformState();
+  }
+
+  // Platform messages are asynchronous, so we initialize in an async method.
+  Future<void> initPlatformState() async {
+    late String edgeVersion;
+    // Platform messages may fail, so we use a try/catch PlatformException.
+    try {
+      edgeVersion = await Edge.extensionVersion;
+    } on PlatformException {
+      log("Failed to get extension versions");
+    }
+
+    // If the widget was removed from the tree while the asynchronous platform
+    // message was in flight, we want to discard the reply rather than calling
+    // setState to update our non-existent appearance.
+    if (!mounted) return;
+
+    setState(() {
+      _edgeVersion = edgeVersion;
+    });
+  }
+
+  Future<void> sendEvent([datasetId]) async {
+    late List<EventHandle> result;
+    Map<dynamic, dynamic> xdmData = {"eventType": "SampleEventType"};
+    Map<String, dynamic> data = {"free": "form", "data": "example"};
+
+    final ExperienceEvent experienceevent = ExperienceEvent({
+      "xdmData": xdmData,
+      "data": data,
+      "datasetIdentifier": datasetId,
+    });
+    try {
+      result = await Edge.sendEvent(experienceevent);
+    } on PlatformException {
+      log("Failed to send experience event");
+    }
```

Comment:
I think we can get this PlatformException removed with the changes in https://github.com/adobe/aepsdk_flutter/pull/3#discussion_r932959640